_posts/2016-06-27-bits-of-shapeless-2.md

```diff
@@ -0,0 +1,519 @@
+---
+layout: post
+title: "Bits of Shapeless part 2: Generic Derivation"
+author: ragb
+---
+
+This is the second instalment in a series of articles about [Shapeless][shapeless].
+The [first post][shapeless1] explained heterogeneous lists (HLists) and how to use them to do type-level recursion. Building on this ground, we will now talk about Generic derivation of case classes and sealed trait hierarchies. Along the way we shall cover other related subjects such as singleton-typed literals, products, coproducts (and labelled versions of those) and shapeless syntax goodies.
+
+At the end of this post you will have built a basic Json rendering library for sealed hierarchies of case classes. Actually, if you aim to be a rockstar Scala coder you have to do a Json library one way or another :).
+
+It is assumed that the reader knows about json, shapeless heterogeneous lists and how to work with them (see my [last post][firstPost]), knows about implicit resolution and is acquainted with the concept of type classes in Scala.
+
+### Setup
+
+Further from [Shapeless][shapeless] off course, we we will use the [JAWN][jawn] Json parsing and `AST` library.
```

Comment:
of course em vez de off course? Não tenho a certeza se querias dizer off course no sentido de fora de percurso mas se sim então está bem, mas ficou um pouco esquisito na minha opinião.